dashboard/test/models/concerns/user/email_preferences_test.rb

```diff
@@ -0,0 +1,154 @@
+require 'test_helper'
+
+class User::EmailPreferencesTest < ActiveSupport::TestCase
```

Comment:
Are you leaving the old tests in place, or do you plan to delete them since we have User::EmailPreferencesTest now?

Reply:
The existing test seems to be more like integration or end-to-end tests since it goes through the entire process of creating a user, which also creates an EmailPReference record.
I tried writing unit tests that focused on more strictly isolated functions. Not sure what to do with the existing tests, I may move them to the integration directory.
https://github.com/code-dot-org/code-dot-org/blob/0a39d109b9540015c6b07fe30097ec480b657f0a/dashboard/test/models/user_test.rb#L2247-L2281

Reply:
I'm going to keep them and create a separate PR to move the integration tests.